src/vehicle.cpp

```diff
@@ -875,6 +875,23 @@ void vehicle::use_controls()
         menu.addentry( release_remote_control, true, 'l', _("Stop controlling") );
     }
 
+    bool has_electronic_controls = false;
+
+    if ( remotely_controlled ) {
+        for( size_t p = 0; p < parts.size(); ++p ) {
+            if (part_flag(p, "CTRL_ELECTRONIC")) {
+                has_electronic_controls = true;
+            }
+        }
+    } else {
+        //TODO: Global coords have to be converted to relative position before it will work
+        std::vector<int> parts_for_check = parts_at_relative( pos_global.x, pos_global.y);
```

Comment:
You can call `vehicle::global_part_at` with a global position instead of `parts_at_relative`.

Or you make the parameter of this function the vehicle part index. Usually, the caller already has that part, it comes for free when looking for the vehicle pointer:
```C++
int part;
vehicle *v = g->m.veh_at(coord, part);
v->use_controls(part);
```
In that case, you can just call `part_with_feature` to check for a specific vehicle part flag.